packages/logger/src/Logger.ts

```diff
@@ -449,6 +449,8 @@ class Logger extends Utility implements LoggerInterface {
               message: UncaughtErrorLogMessage,
               error,
             });
+          } else {
+            loggerRef.clearBuffer();
```

Comment:
I think we could just call `loggerRef.clearBuffer()` in the `finally` block. This way it runs both on errors & regular requests.